src/Microsoft.Azure.SignalR.Common/Endpoints/AadAccessKey.cs

```diff
@@ -65,13 +62,14 @@ public Task<string> GenerateAadToken()
             return await base.GenerateAccessToken(audience, claims, lifetime, algorithm);
         }
 
-        public Task UpdateAccessKeyAsync(IServerNameProvider provider, ILoggerFactory loggerFactory)
+        public async Task UpdateAccessKeyAsync(IServerNameProvider provider, ILoggerFactory factory)
```

Comment:
is this thread safe? can this be called multiple times at the same time?

Reply:
I think it won't happen and the method is thread-safe, only the last time when the method has been called will make changes.